### PR TITLE
refactor(performance): caching, db max-length and async celery

### PR DESCRIPTION
**Pull request by olros**
## Proposed changes
- Made mail-sending async with celery
  - Celery is only used in Production and Development as the tests frozed when celery was used.
- Add caching to permissions and warnings
- Increased max connection length to DB to 60 sec
- Removed the READ_DOT_ENV_FILE variable as it works fine to always read the non-git-checked file and optionally overwriting the defsult values.
- ~~Add pagination to Registrations `/events/<id>/users/`, must be handled frontend before merging this PR~~ Moved to separate issue: #242 
- Added batch sending of emails (max 100 per batch) using "bcc", the recipients won't be able to see each others email-adresses
- Add "TIHLDE" as sender in mails and remove "Hilsen TIHLDE Index" and use only "Hilsen TIHLDE" in email body

Issue number: closes #

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] CHANGELOG.md has been updated. [Guide](https://tihlde.slab.com/posts/changelog-z8hybjom)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Pull request title follows [conventional commits](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) (`type(scope): description`)
- [x] Build (`make PR`) was run locally without failures

## Further comments

This PR is made in order to fix some of the performance issues that where discovered some days ago. In order to test that these changes actually improves the performance I've set up a duplicate of the production server in my own Azure account and runned loadtests on it. The results shows that the number of connections to the database have sunken drastically and the API can handle 16.000 requests in 4 mins with an averge Response time of about 5 secs. Not fantastic, but better. All details can be viewed here: https://loadster.app/dashboard/reports/r0u8VpxSX18ZfPvb.

The caching on user permissions and warnings worked great. I think we can use this in events, news, jobpost, etc and just be good at invalidating the cache on updates. Could maybe write helper methods to ease this.

I did also write a more in-depth guide on how to set up an own version of the Azure hosting on Slab: https://tihlde.slab.com/posts/sett-opp-egen-kopi-av-azure-hosting-fm12c80s

### CODE REVIEW

**Review by Her0elt — APPROVED** [review]
Nice looks like we still have some work to do to completly optimize. Would you post the report in a slack post or something so we have it saved somewhere else

File: app/settings.py [hunk 1]
```diff
@@ -260,11 +256,10 @@
             "filename": "mysite.log",
             "formatter": "verbose",
         },
+        "console": {"class": "logging.StreamHandler", "formatter": "verbose",},
     },
-    "loggers": {
-        "django": {"handlers": ["file"], "propagate": True, "level": "DEBUG",},
-        "MYAPP": {"handlers": ["file"], "level": "DEBUG",},
-    },
+    "loggers": {"django.db.backends": {"propagate": True, "level": "DEBUG",},},
```
Comment (Eirsteir) [inline app/settings.py:261]: Synes ikke vi burde snevre inn loggnivåene på denne måten, da vi kan gå glipp av viktig informasjon hvis det skulle oppstå. Logger fra `django` vil vel også inkludere `django.db.backends` eller er det `root` loggeren som tar det? 

Når jeg tenker over det har vi ingen lett tilgang til disse logfilene, noe som kan være lurt

File: app/common/permissions.py [hunk 2]
```diff
@@ -46,14 +47,20 @@ def has_object_permission(self, request, view, obj):
...
+
     try:
-        user = request.user
-        memberships = user.memberships.all()
-        for membership in memberships:
-            for name in access:
-                if str(membership.group_id).lower() == str(name).lower():
+        CACHE_USER_MEMBERSHIPS_SECONDS = 600
+        CACHE_KEY = f"user__{str(user.user_id)}__memberships"
+        all_memberships = cache.get(CACHE_KEY)
```
Comment (Eirsteir) [inline app/common/permissions.py:57]: Kan være lurt å legge disse variablene i en egen fil slik at vi kan bruke det andre steder når vi trenger det.

**Review by Eirsteir — COMMENTED** [review]
Dette ser veldig bra ut! Gleder meg til å se resultatene fra ny testing. 

Jeg har noen forslag til endringer og spørsmål, og så vil jeg bare minne om denne fra retrospektiven før sommeren: 

> Lage mindre PRs / mindre issues

File: app/settings.py [hunk 3]
```diff
@@ -26,9 +26,7 @@
 PROJECT_ROOT = os.path.dirname(os.path.abspath(__file__))
 DOMAIN = "api.tihlde.org"
 
-READ_DOT_ENV_FILE = os.environ.get("DJANGO_READ_DOT_ENV_FILE", False)
-if READ_DOT_ENV_FILE:
-    load_dotenv(str(BASE_DIR) + "/.env", override=True)
+load_dotenv(str(BASE_DIR) + "/.env", override=True)
```
[the remark this answers is not in the document]
Comment (olros) [reply app/settings.py:29]: Ja

File: app/util/notifier.py [hunk 4]
```diff
@@ -39,23 +45,48 @@ def send_notification(self, title=None, description=None, link=None):
...
+
+    MAX_EMAILS_PER_SENDING = 100
+
+    if (
+        settings.ENVIRONMENT == EnvironmentOptions.PRODUCTION
+        or settings.ENVIRONMENT == EnvironmentOptions.DEVELOPMENT
+    ) and send_async:
+        for mails in chunk_list(to_mails, MAX_EMAILS_PER_SENDING):
+            __send_email.apply_async((mails, html, subject))
+    else:
```
[the remark this answers is not in the document]
Comment (olros) [reply app/util/notifier.py:78]: La inn mulighet for sync av 2 grunner:
- Testene kjører ikke når celery sender epost. Fryser bare. Kan muligens fikses med en mocking av celery/send_email
- Når celery skal sende eposter om for eks evaluering dagen etter et event er det ikke noe poeng å sende dem asynx ettersom det allerede er async

File: app/settings.py (hunk 1 above)
Comment (olros) [reply app/settings.py:261]: Enig, skal bruke `django`. Dette var en glemt endring fra under testing

**Review by Eirsteir — APPROVED** [review]
LGTM

**Pull request merged**